### ISSUE TITLE
NoClassDefFoundError when observing in python Keanu

### ISSUE THREAD

**Issue by johannespetrat** [improbable-research/keanu#450]
**Describe the bug**
There seems to be an error with nd4j that pops up whenever I want to observe a variable. Created a simple program below that observes a Gaussian.

```
Exception in thread "Thread-3" java.lang.ExceptionInInitializerError
	at org.nd4j.linalg.cpu.nativecpu.ops.NativeOpExecutioner.<init>(NativeOpExecutioner.java:71)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.nd4j.linalg.factory.Nd4j.initWithBackend(Nd4j.java:5587)
	at org.nd4j.linalg.factory.Nd4j.initContext(Nd4j.java:5482)
	at org.nd4j.linalg.factory.Nd4j.<clinit>(Nd4j.java:215)
	at io.improbable.keanu.tensor.INDArrayShim.lambda$startNewThreadForNd4j$0(INDArrayShim.java:52)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: ND4J is probably missing dependencies. For more information, please refer to: http://nd4j.org/getstarted.html
	at org.nd4j.nativeblas.NativeOpsHolder.<init>(NativeOpsHolder.java:68)
	at org.nd4j.nativeblas.NativeOpsHolder.<clinit>(NativeOpsHolder.java:36)
	... 11 more
Caused by: java.lang.UnsatisfiedLinkError: no jnind4jcpu in java.library.path
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1867)
	at java.lang.Runtime.loadLibrary0(Runtime.java:870)
	at java.lang.System.loadLibrary(System.java:1122)
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:1258)
	at org.bytedeco.javacpp.Loader.load(Loader.java:999)
	at org.bytedeco.javacpp.Loader.load(Loader.java:891)
	at org.nd4j.nativeblas.Nd4jCpu.<clinit>(Nd4jCpu.java:10)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.bytedeco.javacpp.Loader.load(Loader.java:950)
	at org.bytedeco.javacpp.Loader.load(Loader.java:891)
	at org.nd4j.nativeblas.Nd4jCpu$NativeOps.<clinit>(Nd4jCpu.java:1613)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.nd4j.nativeblas.NativeOpsHolder.<init>(NativeOpsHolder.java:46)
	... 12 more
Caused by: java.lang.UnsatisfiedLinkError: /Users/johannespetrat/.javacpp/cache/nd4j-native-1.0.0-beta3-macosx-x86_64.jar/org/nd4j/nativeblas/macosx-x86_64/libjnind4jcpu.dylib: dlopen(/Users/johannespetrat/.javacpp/cache/nd4j-native-1.0.0-beta3-macosx-x86_64.jar/org/nd4j/nativeblas/macosx-x86_64/libjnind4jcpu.dylib, 1): Library not loaded: @rpath/libmkldnn.0.dylib
  Referenced from: /Users/johannespetrat/.javacpp/cache/nd4j-native-1.0.0-beta3-macosx-x86_64.jar/org/nd4j/nativeblas/macosx-x86_64/./libnd4jcpu.dylib
  Reason: image not found
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
	at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:1238)
	... 23 more
Traceback (most recent call last):
  File "python/black-scholes/sandbox.py", line 15, in <module>
    output.observe(realMu + np.random.randn(1000))
  File "/Users/johannespetrat/.local/share/virtualenvs/applied-keanu-RuPOvGbC/lib/python3.6/site-packages/keanu/vertex/base.py", line 42, in observe
    self.unwrap().observe(Tensor(self.cast(v)).unwrap())
  File "/Users/johannespetrat/.local/share/virtualenvs/applied-keanu-RuPOvGbC/lib/python3.6/site-packages/keanu/tensor.py", line 23, in __init__
    super(Tensor, self).__init__(Tensor.__get_tensor_from_ndarray(t))
  File "/Users/johannespetrat/.local/share/virtualenvs/applied-keanu-RuPOvGbC/lib/python3.6/site-packages/keanu/tensor.py", line 49, in __get_tensor_from_ndarray
    return ctor(values, shape)
  File "/Users/johannespetrat/.local/share/virtualenvs/applied-keanu-RuPOvGbC/lib/python3.6/site-packages/py4j/java_gateway.py", line 1286, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/Users/johannespetrat/.local/share/virtualenvs/applied-keanu-RuPOvGbC/lib/python3.6/site-packages/py4j/protocol.py", line 328, in get_return_value
    format(target_id, ".", name), value)
py4j.protocol.Py4JJavaError: An error occurred while calling z:io.improbable.keanu.tensor.dbl.DoubleTensor.create.
: java.lang.NoClassDefFoundError: Could not initialize class org.nd4j.linalg.factory.Nd4j
	at io.improbable.keanu.tensor.TypedINDArrayFactory.create(TypedINDArrayFactory.java:12)
	at io.improbable.keanu.tensor.dbl.Nd4jDoubleTensor.<init>(Nd4jDoubleTensor.java:58)
	at io.improbable.keanu.tensor.dbl.Nd4jDoubleTensor.create(Nd4jDoubleTensor.java:70)
	at io.improbable.keanu.tensor.dbl.DoubleTensor.create(DoubleTensor.java:40)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
```

**To Reproduce**
```
import keanu as kn
import numpy as np

if __name__ == '__main__':
    mu = kn.vertex.Gaussian(0.0, 0.1)
    output = kn.vertex.Gaussian(mu, 0.1)

    mu.set_label("mu")
    output.set_label("output")

    realMu = 1.
    output.observe(realMu + np.random.randn(1000))
    net = kn.BayesNet(output.get_connected_graph())
```

**Expected behavior**
The numpy array should be converted to a keanu tensor.

**Desktop (please complete the following information):**
 - OS: macOS High Sierra (10.13.3)
 - Keanu Version 0.0.18
 - Python version: 3.6.4
Using pipenv

**Comment by christophernorth** [comment]
Thanks for reporting.  I'm investigating and will update once I know more.

**Comment by christophernorth** [comment]
Seems we have a missing dependency in our Pip package for Keanu.  In the short term, pulling and building the Keanu repo (clone and then ./gradlew build) does seem to resolve the issue for most people.  We're looking in to updating the PIP package.

**Comment by wesleyw72** [comment]
This should be fixed in #498 and will be out in the next release so am closing this issue.
